crates/egui-winit/src/window_settings.rs

```diff
@@ -86,4 +83,58 @@ impl WindowSettings {
             *size = size.at_most(max_size);
         }
     }
+
+    pub fn clamp_window_to_sane_position<E>(
+        &mut self,
+        event_loop: &winit::event_loop::EventLoopWindowTarget<E>,
+    ) {
+        use std::ops::{Add, AddAssign, MulAssign, Sub};
+
+        if let (Some(position), Some(inner_size)) = (&mut self.position, &self.inner_size_points) {
+            let monitors = event_loop.available_monitors();
+            // default to primary monitor, in case the correct monitor was disconnected.
+            let mut monitor: winit::monitor::MonitorHandle = match event_loop.primary_monitor() {
+                Some(m) => m,
+                // No primary monitor, grab index 0 instead
+                _ => {
+                    match event_loop.available_monitors().nth(0) {
+                        Some(m) => m,
+                        None => return, // No monitors?
+                    }
+                }
+            };
+            for _monitor in monitors {
+                let monitor_x_range =
+                    _monitor.position().x..(_monitor.position().x + _monitor.size().width as i32);
+                let monitor_y_range =
+                    _monitor.position().y..(_monitor.position().y + _monitor.size().height as i32);
+
+                if monitor_x_range.contains(&(position.x as i32))
+                    && monitor_y_range.contains(&(position.y as i32))
+                {
+                    monitor = _monitor
+                }
+            }
+
+            let mut inner_size = inner_size.clone(); // Shadow inner_size, we don't want to modify the original
+            inner_size.mul_assign(monitor.scale_factor() as f32);
+            // Add size of title bar. This is 32 px by default in Win 10/11.
+            #[cfg(windows)]
+            inner_size.add_assign(egui::Vec2::new(0.0, 32.0 * monitor.scale_factor() as f32));
```

Comment:
Use proper operators (`+=`) instead of `add_assign` etc

Reply:
For some reason I can't recall, I didn't think that would work. I guess it must have been pretty late in the day by that point. You're absolutely right, that's silly of me.

Reply:
Applied, use of arithmetic functions removed.